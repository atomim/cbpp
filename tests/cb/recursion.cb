#define MINUS(a,b) (a-b)

#define BB 2
#define AA (BB+BB)
#define CC (AA+AA)
#define DD MINUS(AA, BB)

// addtext "magic: " + MINUS(DD, CC)
a = AA + AA
b = AA + CC
c = MINUS(AA,AA)
d = DD
e = MINUS(DD,DD)