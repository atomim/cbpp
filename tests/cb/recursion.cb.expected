// #define MINUS(a,b) (a-b)

// #define BB 2
// #define AA (BB+BB)
// #define CC (AA+AA)
// #define DD MINUS(AA, BB)

// addtext "magic: " + MINUS(DD, CC)
a = (2+2) + (2+2)
b = (2+2) + ((2+2)+(2+2))
c = ((2+2)-(2+2))
d = ((2+2)- 2)
e = (((2+2)- 2)-((2+2)- 2))